Variabilize backdrop class name (#34094)

romdum opened the pull request:
To resolve #34094, create a new class `offcanvas-backdrop` which extends `modal-backdrop`. It helps avoid confusion if an offcanvas and a modal is open and two backdrop are shown.

romdum committed:
Variabilize backdrop class name (#34094)

romdum committed:
Move offcanvas-backdrop css in offcanvas.scss

GeoSot committed:
restrict changes only to backdrop.js, add test

GeoSot commented:
@romdum I did some cleanup in order to help this proceed. I hope you don't mind

@twbs/css-review any help on the sccs? 

The main cause for now is just to extend the `modal-backdrop` class, use it on offcanvas-backdrop, ([initial idea](https://github.com/twbs/bootstrap/pull/34149/commits/df06ad5be12cb8ee15b655081029c6a966b0a48a))

In order to avoid duplication, one Idea is to use a new file `backdrop.scss` create a new class `backdrop` and use it to both as extend

mdo committed:
Add new overlay-backdrop() mixin

- Update modal and offcanvas backdrops to use the mixin (placeholder wouldn't work with params I believe)
- Plus, modals and offcanvases should be able to be used together, so separate values for z-index (already implemented) are reflected properly here.
- Offcanvas still extends modals variables by default, save for z-index

GeoSot committed:
Fix mixin

GeoSot commented:
Js  scoped, seems ok

mdo commented:
Thanks for the fix lol, I must have been braindead earlier. I pushed another change to fix our z-indexes. I'll make sure to cover this in our highlights for the blog post btw. New z-indexes make more sense now IMO.

mdo commented:
I'm on airplane wifi and I screwed up the push, so this was merged to `main`. Working to at least squash those commits in `main` and we can iterate in subsequent PRs if needed. Sorry about that folks!

mdo commented:
I'm on airplane wifi and I screwed up the push, so this was merged to `main`. Working to at least squash those commits in `main` and we can iterate in subsequent PRs if needed. Sorry about that folks!
